modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/DDMFormValuesUtil.java

```diff
@@ -79,6 +84,46 @@ public static DDMFormValues toDDMFormValues(
 		};
 	}
 
+	public static String toText(
+		DDMFormValuesSerializer jsonDDMFormValuesSerializer, DDMForm ddmForm,
+		List<DDMFormFieldValue> ddmFormFieldValues) {
+
+		DDMFormValues ddmFormValues = new DDMFormValues(ddmForm) {
+			{
+				setAvailableLocales(ddmForm.getAvailableLocales());
+				setDDMFormFieldValues(ddmFormFieldValues);
+				setDefaultLocale(ddmForm.getDefaultLocale());
+			}
+		};
+
+		DDMFormValuesSerializerSerializeRequest.Builder builder =
+			DDMFormValuesSerializerSerializeRequest.Builder.newBuilder(
+				ddmFormValues);
+
+		DDMFormValuesSerializerSerializeResponse
+			ddmFormValuesSerializerSerializeResponse =
+				jsonDDMFormValuesSerializer.serialize(builder.build());
+
+		return ddmFormValuesSerializerSerializeResponse.getContent();
+	}
+
+	public static void validateDDMFormValues(
```

Comment:
Can we extract this behavior to a ExceptionMapper and avoid this method? Just call ddmFormValuesValidator.validate from the impl modules

Reply:
Done!